commodore/dependency_mgmt.py

```diff
@@ -82,14 +77,49 @@ def fetch_components(cfg):
     """
 
     click.secho('Discovering components...', bold=True)
-    components = _discover_components(cfg, 'inventory')
+    component_names = _discover_components(cfg, 'inventory')
+    components = _read_component_urls(cfg, component_names)
     click.secho('Fetching components...', bold=True)
     os.makedirs('inventory/classes/components', exist_ok=True)
     os.makedirs('inventory/classes/defaults', exist_ok=True)
     os.makedirs('dependencies/lib', exist_ok=True)
     for c in components:
-        click.echo(f" > {c}...")
-        _fetch_component(cfg, c)
+        if cfg.debug:
+            click.echo(f" > Fetching component {c.name}...")
+        repo = git.clone_repository(c.repository_url, c.target_directory)
+        cfg.register_component(c.name, repo)
+        create_component_symlinks(cfg, c.name)
+
+
+def _read_component_urls(cfg, component_names):
```

Comment:
My personal preference (yes, I'm damaged by C) is to define methods above the point where they're called. I don't mind either way, but right now some methods used by `fetch_components` are defined above the method and this one is defined below.